src/xmipp/libraries/reconstruction/resolution_fso.h

```diff
@@ -0,0 +1,160 @@
+/***************************************************************************
+ *
+ * Authors:    Jose Luis Vilas (joseluis.vilas-prieto@yale.edu)
+ *                             or (jlvilas@cnb.csic.es)
+ *              Hemant. D. Tagare (hemant.tagare@yale.edu)
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+#ifndef _PROG_RES_FSO
+#define _PROG_RES_FSO
+
+#include <core/xmipp_program.h>
+#include <core/xmipp_fftw.h>
+#include <core/matrix2d.h>
+#include <core/xmipp_filename.h>
+#include <core/metadata.h>
+
+
+class ProgFSO : public XmippProgram
+{
+private:
+    // Filenames
+    FileName fnhalf1, fnhalf2, fnmask, fnOut;
+
+    // Double Params
+    double sampling, ang_con, thrs;
+
+    // Int params
+    int Nthreads;
+       
+    // Bool params
+    bool do_3dfsc_filter;
+
+private:
+    //Matrix2d for the projection angles
+    Matrix2D<float> angles;
+
+    // Int params
+    size_t xvoldim, yvoldim, zvoldim, fscshellNum;
+
+	//Frequency vectors and frequency map
+	Matrix1D<double> freq_fourier_x;
+	Matrix1D<double> freq_fourier_y;
+	Matrix1D<double> freq_fourier_z;
+	MultidimArray<float> fx, fy, fz;
+	MultidimArray<float> threeD_FSC, normalizationMap, aniFilter;
+    MultidimArray< double > freqMap;
+
+	//Half maps
+	MultidimArray< std::complex< double > > FT1, FT2;
+	MultidimArray<float> real_z1z2, absz1_vec, absz2_vec;
+
+	//Access indices
+	MultidimArray<long> freqElems, cumpos, freqidx, arr2indx;
+
+
+public:
+        /* The output is a multidim array that define INVERSE of the value of the frequency in Fourier Space
```

Comment:
this method does not have any output (all parameters are constant and return value is void)
maybe say sth like: once done, the result of the computation is stored in this and that class field